src/librustdoc/html/static/main.js

```diff
@@ -782,17 +778,17 @@ function defocusSearchBar() {
                     return 0;
                 });
 
-                var length = results.length;
-                for (i = 0; i < length; ++i) {
-                    var result = results[i];
+                var name, path, parent;
+                for (i = 0, len = results.length; i < len; ++i) {
+                    result = results[i];
 
                     // this validation does not make sense when searching by types
                     if (result.dontValidate) {
                         continue;
                     }
-                    var name = result.item.name.toLowerCase(),
-                        path = result.item.path.toLowerCase(),
-                        parent = result.item.parent;
+                    name = result.item.name.toLowerCase();
+                    path = result.item.path.toLowerCase();
+                    parent = result.item.parent;
```

Comment:
Does this really have a performance difference?

Reply:
No, it's more about cleanup. Recreating the variable at each iteration isn't "clean" (and invalid too, but js doesn't really care apparently).

Reply:
javascript variables defined with `var` are hoisted to the top of the function you're in, so declaring them earlier doesn't make a difference

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Hoisting